test: [M3-7053] - Add Cypress integration test for VPC create flow

jdamore-linode opened the pull request:
## Description 📝
Adds a Cypress integration test for the VPC create flow.

## Major Changes 🔄
```
   - Confirms VPC creation flow using mock API data.
   - Confirms that users can add and remove subnets during create flow.
   - Confirms client side validation when entering invalid labels and IP ranges.
   - Confirms that UI handles API errors gracefully.
   - Confirms that UI redirects to created VPC page after creating a VPC.
```

## How to test 🧪
We can refer to the automated test run, but to run the tests locally you can use this command:

```
yarn cy:run -s "cypress/e2e/core/vpc/vpc-create.spec.ts"
```

hana-linode reviewed (approved):
Ran the tests locally and they passed 🎉

coliu-akamai commented on packages/manager/cypress/e2e/core/vpc/vpc-create.spec.ts:
```diff
@@ -0,0 +1,271 @@
+      .click()
+      .type(`${vpcRegion.label}{enter}`);
+
+    cy.findByText('VPC label').should('be.visible').click().type(mockVpc.label);
```
+1 makes sense to me too -- I'll ask UX to double check :D

coliu-akamai replied:
confirmed that label should be capitalized :D

jdamore-linode replied:
Thanks for confirming @coliu-akamai! I'll make those changes in this PR if that works for everyone 👍

coliu-akamai replied:
thank you!

coliu-akamai reviewed (approved):
✅ confirmed tests pass. Thanks Joe!

jdamore-linode commented on packages/manager/cypress/e2e/core/vpc/vpc-create.spec.ts:
```diff
@@ -0,0 +1,271 @@
+
+    const ipValidationErrorMessage = 'The IPv4 range must be in CIDR format';
+    const vpcCreationErrorMessage = 'An unknown error has occurred.';
+    const totalSubnetUniqueLinodes = mockSubnets.reduce(
+      (acc: number, cur: Subnet) => acc + cur.linodes.length,
+      0
+    );
```
Thanks @hana-linode! I'll take a look at that and see if I can use it from the test

coliu-akamai commented:
Hey @jdamore-linode! Just wanted to give a heads-up that the VPC Create flow will be changed a from this PR/ticket:
https://github.com/linode/manager/pull/9751

The changes are:
- All subnets have the remove [x] button, including the first subnet
- The button text for adding subnets is changed from “Add Subnet” to “Add Another Subnet”
  -  If there are no subnets, it's "Add a Subnet"
- There's a divider below each subnet now

jdamore-linode commented:
> Hey @jdamore-linode! Just wanted to give a heads-up that the VPC Create flow will be changed a from this PR/ticket: #9751
> 
> The changes are:
> 
> * All subnets have the remove [x] button, including the first subnet
> * The button text for adding subnets is changed from “Add Subnet” to “Add Another Subnet” (affects line 156, 199 most likely)
>   
>   * If there are no subnets, it's "Add a Subnet"
> * There's a divider below each subnet now

Thanks @coliu-akamai and @hana-linode for the heads up! I'll review #9751 later today, and my plan is to wait for that work to be reviewed and merged, and then I'll circle back here and update this PR to account for these changes and add additional coverage where necessary (being able to delete the first subnet specifically comes to mind)

jdamore-linode commented:
@coliu-akamai @hana-linode Just letting you know I pushed some changes to account for the recent improvements made to the VPC create page:

- Updated tests to account for "Add a Subnet" label change to "Add another Subnet"
- Added a new test to create a VPC without Subnets (confirms user can delete the subnet on the create page, confirms that button label changes to "Add a Subnet", etc.)

I don't know if this warrants a re-review but wanted to give you both a heads up in case you wanted to put some eyes on it!

coliu-akamai commented on packages/manager/src/features/VPCs/VPCCreate/SubnetNode.tsx:
```diff
@@ -63,15 +63,15 @@ export const SubnetNode = (props: Props) => {
             errorText={subnet.labelError}
             inputId={`subnet-label-${idx}`}
-            label="Subnet label"
+            label="Subnet Label"
```
sorry I forgot to mention -- this change will affect a bunch of tests in the `VPCs/VPCCreate` folder and the `SubnetCreateDrawer.test.tsx` file too. It should be a super quick fix (just updating everything to 'Subnet Label' in those tests, but lmk if you've questions!

jdamore-linode replied:
> sorry I forgot to mention -- this change will affect a bunch of tests in the `VPCs/VPCCreate` folder and the `SubnetCreateDrawer.test.tsx` file too. It should be a super quick fix (just updating everything to 'Subnet Label' in those tests, but lmk if you've questions!

Ah, thank you for the heads up! I'll take care of that soon. Shame on me for not considering the unit tests 😅

coliu-akamai replied:
awesome thanks!

coliu-akamai reviewed (approved):
Other than missing a new Tests changeset and the minor issue I commented about, this looks good + the cypress tests passed. Reapproving on those being resolved -- thanks Joe!!

coliu-akamai commented:
@jdamore-linode  Oh one quick thing, I saw that you changed the Subnet delete dialog to be 'Subnet Label' instead of 'Subnet label' too -- this will affect some of the cypress tests too 😅 (vpc-details-page.spec.ts, line 174, line 241 I think)

jdamore-linode commented:
> @jdamore-linode Oh one quick thing, I saw that you changed the Subnet delete dialog to be 'Subnet Label' instead of 'Subnet label' too -- this will affect some of the cypress tests too 😅 (vpc-details-page.spec.ts, line 174, line 241 I think)

Classic. 😅 Thanks Connie!

jdamore-linode commented:
Going to merge this despite the E2E failure. The failing tests are caused by a gateway timeout when attempting to create LKE clusters, so it's not related to the work in this PR.
